py/dml/breaking_changes.py

```diff
@@ -78,288 +89,324 @@ class broken_conditional_is(CompatFeature):
 
-    * Anonymous banks (`bank { ... }`)
 
-    * Unused templates may instantiate non-existing templates
+@change
+class dml12_remove_misc_quirks(BreakingChange):
+    '''
+    DML 1.2 had several quirks that inadvertedly caused it to
+    permit some strange patterns. In DML 1.4 this was cleaned up and
+    some patterns were disallowed. When this change is enabled, the
+    DML 1.4 behaviour applies also in DML 1.2.
 
-    * The same symbol may be used both for a top-level object (`$`
-      scope) and a top-level symbol (non-`$` scope, e.g. `extern`,
-      `constant` or `loggroup`)
+    Examples of forbidden patterns include:
 
+    * `sizeof(typename)` (see `WSIZEOFTYPE`)
+    * `typeof` on non-lvalue expressions
+    * `select` statements over `vect` types
+    * Passing a string literal to `char *` arguments (now requires `const char *`)
+    * Using the character `-` in the `c_name` parameter of `interface` objects
+    * Overriding interface type via the `c_name` parameter in `implement` objects
+    * `loggroup` identifiers accessible under same name in generated C
+    * Applying `&` to non-lvalues
+    * `extern` statements without type (`extern foo;`)
+    * Anonymous banks (`bank { ... }`)
+    * Unused templates instantiating non-existing templates
+    * Using the same symbol for both top-level object (`$` scope) and
+      top-level symbol (non-`$` scope, e.g. `extern`, `constant` or `loggroup`)
     '''
-    short = "Enable various legacy DML 1.2 quirks"
-    last_api_version = api_6
+    short = "Disallow legacy DML 1.2 quirks"
+    required_after = api_6
 
 
-@feature
-class dml12_goto(CompatFeature):
-    '''The `goto` statement is deprecated; this compatibility feature
-    preserves it. Most `goto` based control structures can be reworked by
-    changing the `goto` into a `throw`, and its label into a `catch`
-    block; since this is sometimes nontrivial, it can be useful to disable
-    the `goto` statement separately.
+@change
+class dml12_remove_goto(BreakingChange):
+    '''
+    Up to Simics 6, the `goto` statement was allowed in DML 1.2.
+    The statement is not allowed in DML 1.4. When this
+    change is enabled, `goto` is disallowed also in DML 1.2.
+    Most `goto`-based control
+    structures can be rewritten using `throw` and `catch` blocks.
     '''
-    short = "Enable the goto statement in DML 1.2"
-    last_api_version = api_6
+    short = "Disallow the goto statement in DML 1.2"
+    required_after = api_6
 
 
-@feature
-class dml12_int(CompatFeature):
-    '''This compatibility feature affects many semantic details of
-    integer arithmetic. When this feature is enabled, DMLC translates
-    most integer operations directly into C, without compensating for
-    DML-specifics, like the support for odd-sized
-    <tt>uint<i>NN</i></tt> types; this can sometimes have unexpected
-    consequences. The most immediate effect of disabling this feature
+@change
+class dml12_fix_int_quirks(BreakingChange):
```

Comment:
_Fix_ int quirks? That's a strange way of putting it, when the intent is rather to strip away strange legacy behaviors. Not blocking, but I encourage you to give naming another think.